modules/mysql/main.tf

```diff
@@ -59,6 +59,12 @@ resource "google_sql_database_instance" "default" {
   lifecycle {
     ignore_changes = ["disk_size"]
   }
+
+    timeouts {
+    create = "${var.create_timeout}"
+    update = "${var.update_timeout}"
+    delete = "${var.delete_timeout}"
+  }
 }
```

Comment:
Have you checked `modules/{mysql,postgresql}/{read_replica,failover_replica}.tf`?
If you would like to manage the timeout of heavy processing, you should specify the `timeouts` configuration block in other `google_sql_instance` resources as well.